src/irc/message.rs

```diff
@@ -4,27 +4,35 @@ pub struct Message<'a> {
   pub params: Vec<&'a str>,
 }
 
-impl Message<'_> {
-  pub fn from_string<'m>(line: &'m str) -> Result<Message<'m>, std::io::Error> {
+impl<'m> Message<'m> {
+  pub fn from_string(line: &'m str) -> Option<Message<'m>> {
     let mut parts = line.split(' ');
-    let prefix = if line.get(0..1).unwrap() == ":" {
-      parts.next()
-    } else {
-      None
-    };
 
-    Ok(Message {
+    let prefix = line.chars().nth(0)
+        .and_then(|c| {
+            if c == ':' {
+                parts.next()
+            } else {
+                None
+            }
+        });
+
+    let command = parts.next()?;
```

Comment:
I thought to use `?` the function has to return a Result..?

Reply:
As of now on stable, `?` can be used with `Option`. https://blog.rust-lang.org/2017/11/22/Rust-1.22.html.

For returning an `Option` into a `Result`, there's this on [nightly](https://doc.rust-lang.org/std/option/struct.NoneError.html), or you can use [ok_or](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or)